### PR TITLE
chore: migrate locale test to TypeScript

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

**Review by naomi-lgbt — APPROVED** [review]
Not sure what's up with the CodeSee map, but this seems fine.

File: config/i18n/all-langs.ts [hunk 1]
```diff
@@ -148,7 +148,7 @@ export const i18nextCodes = {
 };
 
 // These are for the language selector dropdown menu in the footer
-export const langDisplayNames = {
+export const langDisplayNames: Record<string, string> = {
```
Comment (ojeytonwilliams) [inline config/i18n/all-langs.ts:151]:
```suggestion
export const langDisplayNames = {
```
This change would reduce type safety.  There's a few ways to go here, but the DRYest approach I can think of is to use an enum like this

```ts
export enum LangNames {
  English = 'english',
  Espanol = 'espanol',
  ...
}
```
And then wherever we have the string `'english'` we'd have `LangNames.English`.  This should cut down on the number of hoops we need to jump through to convince TS we know what we're doing.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Thanks for taking this on, @Sembauke, it's great that we're taking one more step on the road to 100% TypeScript.

I think we can improve type safety and make @ShaunSHamilton happy (it involves an enum):

**Comment by Sembauke** [comment]
Will open a new pr soon for this

**Pull request closed without merging**